ml-agents/mlagents/trainers/ppo/optimizer_torch.py

```diff
@@ -25,7 +26,21 @@ def __init__(self, policy: TorchPolicy, trainer_settings: TrainerSettings):
         # Create the graph here to give more granular control of the TF graph to the Optimizer.
 
         super().__init__(policy, trainer_settings)
-        params = list(self.policy.actor_critic.parameters())
+        reward_signal_configs = trainer_settings.reward_signals
+        reward_signal_names = [key.value for key, _ in reward_signal_configs.items()]
+
+        if policy.shared_critic:
+            self.value_net = policy.actor
```

Comment:
rename to _value_net (private property) or even self._critic since it is used as critic.